lndclient.go

```diff
@@ -255,6 +255,48 @@ func (l *lndclientGrpc) listChannels() (map[uint64]*channel, error) {
 	return chans, nil
 }
 
+func (l *lndclientGrpc) listClosedChannels() (map[uint64]*channel, error) {
+	ctx, cancel := context.WithTimeout(ctxb, rpcTimeout)
+	defer cancel()
+
+	resp, err := l.main.ClosedChannels(ctx, &lnrpc.ClosedChannelsRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	chans := make(map[uint64]*channel)
+	for _, rpcChan := range resp.Channels {
+		peer, err := route.NewVertexFromStr(rpcChan.RemotePubkey)
+		if err != nil {
+			return nil, err
+		}
+
+		channel := &channel{
+			peer: peer,
+		}
+
+		// LND didn't always store who initiated the channel, so in some cases
+		// we don't know who initiated the channel (for very old channels). We're
+		// unlikely to hit this case since we're dealing with channels related
+		// to current forwards, so we just log that we don't know this value and
+		// allow initiator to be false.
```

Comment:
Agree that this is a very edgy case. But still it is safer to assume initiator to be true I think, because then we don't risk increasing the close cost if we do turn out to be the initiator?

Reply:
> don't risk increasing the close cost if we do turn out to be the initiator

Not sure what you mean here? 

I chose false so that we'd assume they opened the channel and [queue hltcs](https://github.com/lightningequipment/circuitbreaker/blob/master/peer_controller.go#L327).

Reply:
>I chose false so that we'd assume they opened the channel and [queue hltcs](https://github.com/lightningequipment/circuitbreaker/blob/master/peer_controller.go#L327).

Yes, this is the risky part. Suppose that we queue htlcs on a channel that it turns out we are the initiator of. Then if there's a force-close it potentially costs us a lot money. 

If we assume we are the initiator, we won't queue htlcs and don't run that risk.

Reply:
Ah gotcha, hadn't fully understood the queue feature 👍